### ISSUE TITLE
should ur_program_binary_type_t match _pi_program_binary_type ?

### ISSUE THREAD

**Issue by jandres742** [oneapi-src/unified-runtime#804]
_pi_program_binary_type is defined as:

```cpp
    typedef enum {
      PI_PROGRAM_BINARY_TYPE_NONE = 0x0,
      PI_PROGRAM_BINARY_TYPE_COMPILED_OBJECT = 0x1,
      PI_PROGRAM_BINARY_TYPE_LIBRARY = 0x2,
      PI_PROGRAM_BINARY_TYPE_EXECUTABLE = 0x4
    } _pi_program_binary_type;
```

These are the same values for the CL_ counterparts.

   but  ur_program_binary_type_t is defined as:

```cpp
    typedef enum ur_program_binary_type_t {
        UR_PROGRAM_BINARY_TYPE_NONE = 0,            ///< No program binary
    is associated with device
        UR_PROGRAM_BINARY_TYPE_COMPILED_OBJECT = 1, ///< Program binary is
    compiled object
        UR_PROGRAM_BINARY_TYPE_LIBRARY = 2,         ///< Program binary is
    library object
        UR_PROGRAM_BINARY_TYPE_EXECUTABLE = 3,      ///< Program binary is
    executable
        /// @cond
        UR_PROGRAM_BINARY_TYPE_FORCE_UINT32 = 0x7fffffff
        /// @endcond

    } ur_program_binary_type_t;
```

so PI_PROGRAM_BINARY_TYPE_EXECUTABLE is 4 but UR_PROGRAM_BINARY_TYPE_EXECUTABLE  is 3. 

we could change UR_PROGRAM_BINARY_TYPE_EXECUTABLE   to 4, so it matches the PI_ and CL_values, or we can rely on adapters to do the change. Advantage of the former is that we would avoid problems with other parts of SYCL runtime and libraries that assume already the same numbering between PI and CL.

**Comment by veselypeta** [comment]
I think this should be handled in the pi2ur layer. We actually do some conversions for this already, i.e. https://github.com/intel/llvm/blob/sycl/sycl/plugins/unified_runtime/pi2ur.hpp#L4178. Perhaps we should extend this for when querying the binary type

**Comment by jandres742** [comment]
thanks @veselypeta . That was another option I was thinking.

**Comment by kbenzie** [comment]
Seems UR is the odd one out here, since PI matches the [OpenCL-Headers definitions](https://github.com/KhronosGroup/OpenCL-Headers/blob/main/CL/cl.h#L736-L740) for these values.

**Comment by jandres742** [comment]
@kbenzie so do you think it is better to have a match of enum values instead of handling the conversion in the pi2ur as proposed in https://github.com/intel/llvm/pull/10828?

**Comment by kbenzie** [comment]
In isolation I'd say its better they match. However I think it will be more disruptive to change the UR spec/headers than make the change in pi2ur.

**Comment by jandres742** [comment]
@kbenzie if we think it is better to match, it is better to do it now. We dont have yet 1.0, so breaking compatibility or noise on headers is not that disruptive. Let's discuss it in the next WG.

**Comment by alycm** [comment]
Discussed 2023-08-23: Will check if translation in pi2ur is enough for current release.  Do not change the UR enum values after that.  The OpenCL adapter will need checked in case it is assuming that the UR enum values match OpenCL.

**Comment by jandres742** [comment]
No concerns from reporter. Doing translation in UR layer is good. We should proceed then with https://github.com/intel/llvm/pull/10828

**Comment by veselypeta** [comment]
Closed with intel/llvm#10828
